homeassistant/components/incomfort/config_flow.py

```diff
@@ -102,6 +104,20 @@ def async_get_options_flow(
         """Get the options flow for this handler."""
         return InComfortOptionsFlowHandler()
 
+    async def async_step_dhcp(
+        self, discovery_info: DhcpServiceInfo
+    ) -> ConfigFlowResult:
+        """Prepare configuration for a DHCP discovered Intergas Gateway device."""
+        host = discovery_info.ip
+        unique_id = format_mac(discovery_info.macaddress)
+        await self.async_set_unique_id(unique_id)
+        self._abort_if_unique_id_configured()
+
+        await self.async_set_unique_id(unique_id)
+        self._abort_if_unique_id_configured(updates={CONF_HOST: host})
+
+        return await self.async_step_user({CONF_HOST: host})
```

Comment:
It looks like they would be allowed to enter another host if it fails to connect which means the unique id would be wrong if they are able to successfully set up another host

Reply:
Added validation